packages/expo-modules-core/ios/Swift/Methods/AnyArgumentType.swift

```diff
@@ -0,0 +1,43 @@
+
+struct AnyArgumentType: CustomDebugStringConvertible {
+  let baseType: Any.Type
+
+  private let castHelper: (Any) -> AnyMethodArgument?
+  private let canCastHelper: (Any) -> Bool
+
+  init<T: AnyMethodArgument>(_ baseType: T.Type) {
+    self.baseType = baseType
+
+    self.castHelper = { $0 as? T }
+
+    // handle class metatypes separately in order to allow T.self != base.
+    if baseType is AnyClass {
+      self.canCastHelper = { x in
+        sequence(
+          first: Mirror(reflecting: x), next: { $0.superclassMirror }
+        )
+        .contains { $0.subjectType == baseType }
+      }
+    } else {
+      self.canCastHelper = { $0 is T }
```

Comment:
Maybe it isn't a problem, but I want to be sure. `castHelper` try to safe cast to `T?`. On the other hand, `canCastHelper` check if a cast is possible to `T`. This doesn't matter if T is nullable, but what happens if T isn't, but the user provided null?

Reply:
Actually it's not possible to call it with `nil`
- `canCastHelper` takes an argument of `Any` type, not `Any?`
- in `canCast<T>(_ object: T) -> Bool` the type couldn't be inferred if you pass `nil`
- generic types are always non-optional